test/token/ERC721/ERC721PausableToken.test.js

```diff
@@ -0,0 +1,32 @@
+const { shouldBehaveLikeERC721PausableToken } = require('./ERC721PausableToken.behavior');
+const { shouldBehaveLikeERC721BasicToken } = require('./ERC721BasicToken.behavior');
+
+const BigNumber = web3.BigNumber;
+const ERC721PausableToken = artifacts.require('ERC721PausableTokenMock.sol');
+
+require('chai')
+  .use(require('chai-bignumber')(BigNumber))
+  .should();
+
+contract('ERC721PausableToken', function ([_, owner, recipient, operator, ...otherAccounts]) {
+  beforeEach(async function () {
+    this.token = await ERC721PausableToken.new({ from: owner });
+  });
+
+  context('when token is paused', function () {
```

Comment:
This looks a bit weird, since the test does not pause, it is the behavior that calls the `pause` function. Shouldn't we have the `beforeEach` here pausing the token?

Reply:
Should the behavior name be changed from `shouldBehaveLikeERC721PausableToken` to `shouldBehaveLikeERC721PausedToken` then?

Or maybe just leave 'when token is paused and then unpaused' context, and move out two first entries (as they don't have beforeEach'es).

Reply:
I'm not sure what you mean by 'moving out the two first entries' (which?), but the behavior renaming sounds like a neat idea!

Reply:
I mean just to remove two first `context` blocks:

```
shouldBehaveLikeERC721BasicToken(owner, [...otherAccounts]);
shouldBehaveLikeERC721PausableToken(owner, [...otherAccounts]);

context('when token is paused and then unpaused', function () {
   beforeEach(async function () {
      await this.token.pause({ from: owner });
      await this.token.unpause({ from: owner });
   });

   shouldBehaveLikeERC721BasicToken(owner, [...otherAccounts]);
});
```

But I'm not shure which option is more prefferable.

Reply:
I'd go with pausing outside, in a context, and renaming the behavior: in that second example you're testing both behaviors at the same time, as if they were the same thing (you could, in fact, test again for `shouldBehaveLikeERC721PausableToken` after pausing and unpausing :) )